src/NHibernate/Multi/LinqBatchItem.cs

```diff
@@ -20,6 +20,10 @@ public static class LinqBatchItem
 	{
 		internal static ConcurrentDictionary<System.Type, Func<ILinqBatchItem, IList>> GetResultsForTypeDic = new ConcurrentDictionary<System.Type, Func<ILinqBatchItem, IList>>();
 
+		internal static MethodInfo GetTypedResultsMethod = ReflectHelper
+			.GetMethod((ILinqBatchItem i) => i.GetTypedResults<object>())
```

Comment:
Did you test it? I think it would still fail. It wasn't working because it's not generic delegate

Reply:
It should work. It is (almost) the same code as you had.

Reply:
That's delegate you create here:
```C#
public IList MyMethod(ILinqBatchItem i) => i.GetTypedResults<object>();
```
It's not generic. I will fix it soon

Reply:
Hm.. Tests are green...  But delegate always returns `List<object>`. Don't you agree? It's just seems not important for affected queries that return aggregate results.. I'm still going to push fixed changes with your first solution to avoid boxing...

Reply:
Ok.. I'm wrong. I tested it and it works as expected. I misunderstood how `ReflectHelper.GetMethod`.. But strange why `CreateDelegate` didn't work with it

Reply:
I’m calling ``GetMethod works in reverse of `Expression.Call` — it assumes that lambda’s body consists of only `MethodCalExpression` and extracts it’s `MethodInfo`.

In this case the MethodInfo is already closed generic with “object” generic argument, so I’m calling `GetGenericMethodDefinition()`